pkg/promtail/position.go

```diff
@@ -29,14 +29,15 @@ func (cfg *PositionsConfig) RegisterFlags(flags *flag.FlagSet) {
 
 // Positions tracks how far through each file we've read.
 type Positions struct {
-	logger    log.Logger
-	cfg       PositionsConfig
-	mtx       sync.Mutex
-	positions map[string]int64
-	quit      chan struct{}
+	logger       log.Logger
+	cfg          PositionsConfig
+	mtx          sync.Mutex
+	positions    map[string]int64
+	quit         chan struct{}
+	quitComplete chan struct{}
 }
 
-type positionsFile struct {
+type PositionsFile struct {
```

Comment:
Why did you need to expose this?  Capital letters at the beginning of a type make it package-public.

Reply:
because i was loading the positions file in the unit test to verify the contents and debated between just duplicating the struct in the unit test or exposing it here.  I don't have a strong argument for one way vs the other and didn't see the harm in exposing it so picked that.

Reply:
You should be able to use the same type in a unit test, the test should be in the same package.****

Reply:
ahh, you are correct, I had a collision with a variable named positionsFile and misunderstood the error, I will return the struct to private